http/jsonapi/runtime/standart_params.go

```diff
@@ -63,46 +60,86 @@ func (m *MapMapper) Map(value string) (string, bool) {
 	return val, isValid
 }
 
-// PaginationFromRequest extracts pagination query parameter and  returns a function that adds the pagination to a query
-func PaginationFromRequest(r *http.Request) (QueryOption, error) {
+// UrlQueryParameters contains all information that are needed for pagination, sorting and filtering.
+// It is not depending on orm.Query
+type UrlQueryParameters struct {
+	HasPagination bool
+	PageNr        int
+	PageSize      int
+	Order         []string
+	Filter        map[string][]interface{}
+}
+
+// ReadURLQueryParameters reads sorting, filter and pagination from requests and return a UrlQueryParameters object,
+// even if any errors occur. The returned error combines all errors of pagination, filter and sorting.
+func ReadURLQueryParameters(r *http.Request, mapper ColumnMapper, sanitizer ValueSanitizer) (*UrlQueryParameters, error) {
+	result := &UrlQueryParameters{}
+	errPagination := result.setPagination(r)
```

Comment:
minor issue, `setXXX` implies that result does something to the pointer (method argument) it receives. This might be confusing when looking at these methods in the future.

Mabye `readXXX` or `extractXXX`?